artman/conductors/cloudtasks_conductor.py

```diff
@@ -0,0 +1,147 @@
+
+
+def _create_tasks_client():
+    credentials = GoogleCredentials.get_application_default()
+    with open(
+        os.path.join(os.path.dirname(__file__), 'cloudtasks.json'), 'r') as f:
+            return build_from_document(f.read(), credentials=credentials)
+
+
+def _pull_task(task_client, queue_name):
+    body = {
+      "maxTasks": 1,
+      "leaseDuration": {"seconds": 300, "nanos": 0},  # Expire after 300 secs.
+      "responseView": "FULL",
+      "name": "%s" % queue_name
+    }
+    tasks = task_client.projects().locations().queues().tasks().pull(
+        name=queue_name, body=body).execute()
+    logger.info('Pulling tasks request returned %s' % tasks)
+    return tasks
+
+
+def _ack_task(task_client, task):
+    body = {'scheduleTime': task['scheduleTime']}
+    response = task_client.projects().locations().queues().tasks().acknowledge(
+        name=task['name'],
+        body=body).execute()
+    logger.info('Acknowledge task request returned %s' % response)
+    return response
+
+
+def _cancel_task_lease(task_client, task):
+    body = {'scheduleTime': task['scheduleTime'], "responseView": "FULL"}
+    response = task_client.projects().locations().queues().tasks().cancelLease(
+        name=task['name'],
+        body=body).execute()
+    logger.info('Cancel task request returned %s' % response)
+    return response
+
+
+def _execute_task(artman_user_config, task):
+    """Execute the remote artman tasks.
+
+    It execute the artman command with a customized artman user config and
+    additional pipeline arguments."""
+    task_payload = base64.b64decode(task['pullTaskTarget']['payload'])
+    artman_args = task_payload.decode("utf-8").split(' ')
+    artman_args.append('--user-config')
+    artman_args.append(artman_user_config)
+    # TODO(ethanbao): No longer need to override toolkit_path once the
+    # toolkit_path config in artman user config won't be overridden with
+    # ${REPOROOT}/toolkit
+    if os.environ.get('TOOLKIT_HOME'):
+        artman_args.append('--pipeline-kwargs')
+        artman_args.append(
+            '{"toolkit_path": "%s"}' % os.environ.get('TOOLKIT_HOME'))
+    main.main(*artman_args)
+
+
+def _prepare_dir(source_repo="https://github.com/googleapis/googleapis.git"):
+    """Prepare the temporary folder to task execution.
+
+    It downloads the googleapis repo and adds a one-time artman config yaml.
+    TODO(ethanbao): support loading more input files from heterogeneous data
+    sources"""
+
+    repo_root = '/tmp/artman/%s' % str(uuid.uuid4())
+    logger.info('Prepare a temporary root repo: %s' % repo_root)
+    try:
+        os.makedirs(repo_root)
+    except OSError as e:
+        raise e
+    logger.info('Checking out fresh clone of %s.' % source_repo)
+    googleapis_dir = os.path.join(repo_root, "googleapis")
+    git_clone_args = ['git', 'clone', source_repo, googleapis_dir]
+    output = subprocess.check_output(git_clone_args)
+    if output:
```

Comment:
P.S. A potentially useful refactor would be to pull `exec_command` out of `TaskBase` and make it a utility function, and then have it be called instead of directly calling `subprocess.check_output`. This would ensure that the logging is always consistent.

I do _not_ think this PR needs to block on that change, though.